src/apis.cpp

```diff
@@ -127,28 +127,29 @@ static int gds_rollback_qp(struct gds_qp *qp, gds_send_request_t * send_info, en
 //-----------------------------------------------------------------------------
 
 #define ntohll(x) (((uint64_t)(ntohl((int)((x << 32) >> 32))) << 32) |  (uint32_t)ntohl(((int)(x >> 32))))
-static void gds_dump_swr(const char * func_name, struct ibv_qp_swr_info swr_info)
+
+static void gds_dump_swr(const char * func_name, struct gds_swr_info swr_info)
 {
     gds_dbg("[%s] wr_id=%lx, num_sge=%d\n", func_name, swr_info.wr_id, swr_info.num_sge);
 
     for(int j=0; j < swr_info.num_sge; j++)
     {
-        gds_dbg("[%s]    SGE=%d, Size ptr=0x%08x, Size=%d (0x%08x), +offset=%d\n", 
+        gds_dbg("[%s]    SGE=%d, Size ptr=00x%lx, Size=%d (0x%08x), +offset=%d\n",
```

Comment:
There's a typo here (00x), and you might want to put debugging print changes in a separate patch, to make the review easier.